staging/src/k8s.io/client-go/tools/clientcmd/loader.go

```diff
@@ -129,17 +131,33 @@ type ClientConfigLoadingRules struct {
 // ClientConfigLoadingRules implements the ClientConfigLoader interface.
 var _ ClientConfigLoader = &ClientConfigLoadingRules{}
 
+// getRecommendedHomeFileOrDir returns RecommendedHomeFileDir if RecommendedHomeFile is empty or non-existent.
+// otherwise returns RecommendedHomeFile.
+func getRecommendedHomeFileOrDir() []string {
+	if len(RecommendedHomeFile) == 0 {
+		return []string{RecommendedHomeFileDir}
+	}
+	_, err := os.Stat(RecommendedHomeFile)
+	if os.IsNotExist(err) {
+		return []string{RecommendedHomeFileDir}
```

Comment:
Only do this if the directory already exists.  Otherwise prefer the existing behavior.